inc/admin/options.php

```diff
@@ -1344,3 +1309,69 @@ function rocket_database_count_cleanup_items( $type ) {
 
 	return $count;
 }
+
+
+/**
+ * Handle WP Rocket settings import.
+ * 
+ * @since 2.10.7
+ * @author Remy Perona
+ *
+ * @param array $file_import File import data.
+ * @param string $filename_prefix import filename prefix.
+ * @param array $inputs Original settings.
+ * @return bool|array False if there was an error, an array of imported settings otherwise.
+ */
+function rocket_handle_settings_import( $file_import, $filename_prefix, $inputs ) {
+	if ( ! isset( $file_import ) ) {
+		add_settings_error( 'general', 'settings_update_error', __( 'Settings import failed: no file uploaded.', 'rocket' ), 'error' );
+		return false;
+	}
+
+	if ( ! preg_match( '/' . $filename_prefix . '-settings-20\d{2}-\d{2}-\d{2}-[a-f0-9]{13}\.(?:txt|json)/', $file_import['name'] ) ) {
+		add_settings_error( 'general', 'settings_update_error', __( 'Settings import failed: incorrect filename.', 'rocket' ), 'error' );
+		return false;
+	}
+
+	add_filter( 'mime_types', 'rocket_allow_json_mime_type' );
+
+	$file_data = wp_check_filetype_and_ext(  $file_import['tmp_name'], $file_import['name'] );
+
+	if ( 'text/plain' !== $file_data['type'] && 'application/json' !== $file_data['type'] ) {
+		add_settings_error( 'general', 'settings_update_error', __( 'Settings import failed: incorrect filetype.', 'rocket' ), 'error' );
+		return false;
+	}
+
+	$_post_action 		= $_POST['action'];
+	$_POST['action'] 	= 'wp_handle_sideload';
+	$file 				= wp_handle_sideload( $file_import );
+	$_POST['action'] 	= $_post_action;
+	$settings 			= rocket_direct_filesystem()->get_contents( $file['file'] );
+	remove_filter( 'mime_types', 'rocket_allow_json_mime_type' );
+	
+	if ( 'text/plain' === $file_data['type'] ) {
+		$gz 				= 'gz' . strrev( 'etalfni' );
+		$settings 			= $gz// ;
+		( $settings );
+		$settings 			= unserialize( $settings );
+	} elseif ( 'application/json' === $file_data['type'] ) {
+		$settings = (array) json_decode( $settings );
```

Comment:
It's a detail, but if we test `is_array( $settings )` on L1364, this `(array)` is useless.